### PR TITLE
Moving from Codenvy CI to the CentOS CI: Add sripts for CI Jobs

### PR DESCRIPTION
**Pull request by vparfonov**
Signed-off-by: Vitalii Parfonov <vparfono@redhat.com>

### What does this PR do?

Add scripts for:

- _each commits builds_: will build and push artifacts to the `https://oss.sonatype.org/#welcome`;
- _nightly builds_: will build and push artifacts to the nexus, build and push docker images to the `https://quay.io/repository/eclipse` with `nightly` tags
- _release script_ (not tested yet)

### CODE REVIEW

File: .ci/cico_common.sh [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    eclipse/che-postgres
+    eclipse/che-dev
+    eclipse/che-server
+    eclipse/che-dashboard-dev
+    eclipse/che-e2e
+    )
+
+
+    # BUILD IMAGES
+    for image_dir in ${DOCKER_FILES_LOCATIONS[@]}
```
Comment (ibuziuk) [inline .ci/cico_common.sh:118]: would be nice to have this as a separate function IMO

File: .ci/cico_common.sh [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+         fi
+         if [ $? -ne 0 ]; then
+           echo "ERROR:"
+           echo "build of '$image_dir' image is failed!"
+           exit 1
+         fi
+     done
+
+    #PUSH IMAGES
+    for image in ${IMAGES_LIST[@]}
```
Comment (ibuziuk) [inline .ci/cico_common.sh:139]: would be nice to have this as a separate function IMO

**Review by ibuziuk — APPROVED** [review]
LGTM

File: .ci/cico_common.sh [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    echo $CHE_OSS_SONATYPE_GPG_KEY | base64 -d > $HOME/.m2/gpg.key
+    set -x
+    gpg --import $HOME/.m2/gpg.key
+}
+
+install_deps(){
+    set +x
+    yum -y update
+    yum -y install centos-release-scl-rh java-1.8.0-openjdk-devel git 
+    yum -y install rh-maven33
```
Comment (skabashnyuk) [inline .ci/cico_common.sh:41]: can we get a bit newer maven? 3.6.3?

File: .ci/cico_common.sh [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        exit 1
+    fi
+}
+
+releaseProject() {
+    #test 4
+    CUR_VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+    TAG = $(echo $CUR_VERSION | cut -d'-' -f1) #cut SNAPSHOT form the version name
+    echo -e "\x1B[92m############### Release: $TAG\x1B[0m"
+    scl enable rh-maven33 "mvn release:prepare release:perform -B -Dresume=false -Dtag=$TAG -DreleaseVersion=$TAG '-Darguments=-DskipTests=true -Dskip-validate-sources -Dgpg.passphrase=$CHE_OSS_SONATYPE_PASSPHRASE -Darchetype.test.skip=true -Dversion.animal-sniffer.enforcer-rule=1.16'"
```
Comment (skabashnyuk) [inline .ci/cico_common.sh:68]: why dont we change `version.animal-sniffer.enforcer-rule` in upstream?

**Review by skabashnyuk — APPROVED** [review]

**Comment by vparfonov** [comment]
I will merge it as is at the moment. Will try to provide suggested improvements in next PR.

**Pull request merged**